src/common/downloadFileDialog.component.tsx

```diff
@@ -0,0 +1,99 @@
+import WarningIcon from '@mui/icons-material/Warning';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormHelperText,
+} from '@mui/material';
+import { UseQueryResult } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import React from 'react';
+import { APIImage, APIImageWithURL } from '../api/api.types';
+import { downloadFileByLink } from '../utils';
+
+export interface BaseDownloadFileProps {
+  open: boolean;
+  onClose: () => void;
+  fileType: 'Image' | 'Attachment';
+}
+
+export interface ImageDownloadDialogProps extends BaseDownloadFileProps {
+  fileType: 'Image';
+  useGetFile: (id: string) => UseQueryResult<APIImageWithURL, AxiosError>;
+  file: APIImage;
+}
+
+export type DownloadFileProps = ImageDownloadDialogProps;
+
+const DownloadFileDialog = (props: DownloadFileProps) => {
+  const { open, onClose, fileType, file, useGetFile } = props;
+
+  const [formError, setFormError] = React.useState<string | undefined>(
+    undefined
+  );
+
+  const { data: downloadedFile, isLoading: downloadIsLoading } = useGetFile(
+    file.id
+  );
+
+  const handleClose = React.useCallback(() => {
+    setFormError(undefined);
+    onClose();
+  }, [onClose]);
+
+  const handleDownloadFile = React.useCallback(() => {
+    if (downloadedFile) {
+      downloadFileByLink(document, downloadedFile);
+      onClose();
+    } else {
+      setFormError('No data provided. Please refresh and try again');
+    }
+  }, [downloadedFile, onClose]);
+
+  return (
+    <Dialog open={open} maxWidth="lg">
+      <DialogTitle
+        sx={{ display: 'inline-flex', alignItems: 'center' }}
+        data-testid="download-file-dialog-title"
+      >
+        <WarningIcon sx={{ marginRight: 1 }} />
+        Download {fileType}?
+      </DialogTitle>
+      <DialogContent data-testid="download-images-message">
+        Are you sure you want to download{' '}
+        <strong data-testid="download-images-value">{file.file_name}</strong>?
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={handleClose}>Cancel</Button>
+        <Button
+          onClick={handleDownloadFile}
```

Comment:
The onClick should trigger the request for the download_url (GET request), instead of fetching it immediately (as it currently does on Line 39). Fetching it straight away can cause issues since the download_url has an expiry time. Additionally, it could result in unnecessary requests to the backend if the user accidentally leaves the dialog open.